### ISSUE TITLE
Add support for stopping criteria other than training iterations in TensorFlowTransform.

### ISSUE THREAD

**Issue by zeahmed** [dotnet/machinelearning#1108]
Currently, training in TensorFlowTransform stops when specified umber of iterations are done. 
It will also be interesting to add support for following stopping criteria.
- Increase in training loss (or no change in the loss over the last few iterations)
- Stop training based on performance on validation set.
